Invoke everything in Tools via Python 3

gsnedders opened the pull request:
#### b8af722e543ceb489b3daebef9c18689c9adbb27
<pre>
Invoke everything in Tools via Python 3
<a href="https://bugs.webkit.org/show_bug.cgi?id=260916">https://bugs.webkit.org/show_bug.cgi?id=260916</a>

Reviewed by Aakash Jain.

Python 2 should be dead already. This shouldn&apos;t regress anything,
given most of these scripts are in practice already invoked via
`python3` explicitly.

Bot-watchers: If you&apos;re reading this, I&apos;m sorry. Please file a bug for
whatever this broke and just revert this and ensure any services that
regressed get redeployed.

* Tools/CISupport/build-webkit-org/steps.py:
(InstallWinCairoDependencies):
* Tools/Scripts/bisect-builds:
(download_archive):
(extract_archive):
* Tools/Scripts/build-webkit:
* Tools/Scripts/check-for-platform-layering-violations:
* Tools/Scripts/check-inspector-strings:
* Tools/Scripts/compare-results:
* Tools/Scripts/compare-webkit-configurations:
* Tools/Scripts/download-github-release.py:
* Tools/Scripts/dump-webkit-tests-run:
* Tools/Scripts/export-w3c-performance-wg-tests:
* Tools/Scripts/export-w3c-test-changes:
* Tools/Scripts/find-duplicate-files:
* Tools/Scripts/generate-coverage-data:
(generateReport):
* Tools/Scripts/import-w3c-performance-wg-tests:
* Tools/Scripts/import-webdriver-tests:
* Tools/Scripts/malloc-tree:
* Tools/Scripts/mark-jsc-stress-test:
* Tools/Scripts/merge-result-jsons:
* Tools/Scripts/parse-gc-phase-timings:
* Tools/Scripts/read-checksum-from-png:
* Tools/Scripts/rebase-patch-after-webkit-move:
* Tools/Scripts/run-web-platform-tests:
* Tools/Scripts/run-webkit-websocketserver:
(openWebSocketServer):
(closeWebSocketServer):
* Tools/Scripts/sampstat:
* Tools/Scripts/sync-feature-defines:
* Tools/Scripts/update-test-expectations-from-bugzilla:
* Tools/Scripts/update-webkit-wincairo-libs.py:
* Tools/Scripts/webkitperl/BuildSubproject.pm:
* Tools/Scripts/webkitpy/tool/gcovr:
* Tools/lldb/dump_class_layout_unittest.py:
* Tools/lldb/lldb_webkit_unittest.py:

Canonical link: <a href="https://commits.webkit.org/267608@main">https://commits.webkit.org/267608@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bde066886d3e925a58596e36363830e5717d1bdb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16875 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/17198 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17654 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18657 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15802 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/17065 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20430 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17336 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18063 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17074 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17433 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14611 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19463 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/17005 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14676 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/15298 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/22038 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/14561 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15666 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15465 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/19783 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/16100 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16074 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13628 "1 failures") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/18431 "Built successfully") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/16743 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15243 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/4306 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4096 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19606 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/19657 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15905 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/4159 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-app/manage.py:
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
ews-app doesn't have proper python3 support yet. The service is still running using Python2. It would be moved to Python3 separately. Please exclude this file from this PR.

aj062 reviewed (approved):
rs=me
Can consider breaking and landing the PR in multiple parts as well.

gsnedders committed:
Invoke everything in Tools via Python 3
https://bugs.webkit.org/show_bug.cgi?id=260916

Reviewed by Aakash Jain.

Python 2 should be dead already. This shouldn't regress anything,
given most of these scripts are in practice already invoked via
`python3` explicitly.

Bot-watchers: If you're reading this, I'm sorry. Please file a bug for
whatever this broke and just revert this and ensure any services that
regressed get redeployed.

* Tools/CISupport/build-webkit-org/steps.py:
(InstallWinCairoDependencies):
* Tools/Scripts/bisect-builds:
(download_archive):
(extract_archive):
* Tools/Scripts/build-webkit:
* Tools/Scripts/check-for-platform-layering-violations:
* Tools/Scripts/check-inspector-strings:
* Tools/Scripts/compare-results:
* Tools/Scripts/compare-webkit-configurations:
* Tools/Scripts/download-github-release.py:
* Tools/Scripts/dump-webkit-tests-run:
* Tools/Scripts/export-w3c-performance-wg-tests:
* Tools/Scripts/export-w3c-test-changes:
* Tools/Scripts/find-duplicate-files:
* Tools/Scripts/generate-coverage-data:
(generateReport):
* Tools/Scripts/import-w3c-performance-wg-tests:
* Tools/Scripts/import-webdriver-tests:
* Tools/Scripts/malloc-tree:
* Tools/Scripts/mark-jsc-stress-test:
* Tools/Scripts/merge-result-jsons:
* Tools/Scripts/parse-gc-phase-timings:
* Tools/Scripts/read-checksum-from-png:
* Tools/Scripts/rebase-patch-after-webkit-move:
* Tools/Scripts/run-web-platform-tests:
* Tools/Scripts/run-webkit-websocketserver:
(openWebSocketServer):
(closeWebSocketServer):
* Tools/Scripts/sampstat:
* Tools/Scripts/sync-feature-defines:
* Tools/Scripts/update-test-expectations-from-bugzilla:
* Tools/Scripts/update-webkit-wincairo-libs.py:
* Tools/Scripts/webkitperl/BuildSubproject.pm:
* Tools/Scripts/webkitpy/tool/gcovr:
* Tools/lldb/dump_class_layout_unittest.py:
* Tools/lldb/lldb_webkit_unittest.py:

Canonical link: https://commits.webkit.org/267608@main
